src/Tribe/Tickets.php

```diff
@@ -1051,73 +1051,87 @@ public static function add_frontend_stock_data( array $tickets ) {
-			if ( ! $tickets ) {
-				return array();
-			}
-
-			/**
-			 * This order is important so we that tickets overwrite RSVP on
-			 * the Buy Now Button on the front-end
-			 */
-			$types['rsvp']    = array(
-				'count'     => 0,
-				'stock'     => 0,
-				'unlimited' => 0,
-				'available' => 0,
-			);
-			$types['tickets'] = array(
-				'count'     => 0, // count of tickets currently for sale
-				'stock'     => 0, // current stock of tickets available for sale
-				'global'    => 0, // global stock ticket
-				'unlimited' => 0, // unlimited stock tickets
-				'available' => 0, // are tickets available for sale right now
-			);
-
-			foreach ( $tickets as $ticket ) {
-				// If a ticket is not current for sale do not count it
-				if ( ! tribe_events_ticket_is_on_sale( $ticket ) ) {
-					continue;
-				}
-
-				// if ticket and not rsvp add to ticket array
-				if ( 'Tribe__Tickets__RSVP' !== $ticket->provider_class ) {
-					$types['tickets']['count'] ++;
-
-					if ( Tribe__Tickets__Global_Stock::GLOBAL_STOCK_MODE === $ticket->global_stock_mode() && 0 === $types['tickets']['global'] ) {
-						$types['tickets']['global'] ++;
-					} elseif ( Tribe__Tickets__Global_Stock::GLOBAL_STOCK_MODE === $ticket->global_stock_mode() && 1 === $types['tickets']['global'] ) {
-						continue;
-					}
-
-					$types['tickets']['stock'] = $types['tickets']['stock'] + $ticket->stock;
-
-					if ( 0 !== $types['tickets']['stock'] ) {
-						$types['tickets']['available'] ++;
-					}
-
-					if ( ! $ticket->manage_stock() ) {
-						$types['tickets']['unlimited'] ++;
-						$types['tickets']['available'] ++;
-					}
-				} else {
-					$types['rsvp']['count'] ++;
+		    $tickets = self::get_all_event_tickets( $event_id );
+
+		    // if no tickets or rsvp return empty array
+		    if ( ! $tickets ) {
+			return array();
+		    }
+
+		    /**
+		     * This order is important so we that tickets overwrite RSVP on
+		     * the Buy Now Button on the front-end
+		     */
+		    $types['rsvp']    = array(
+			'count'     => 0,
+			'stock'     => 0,
+			'unlimited' => 0,
+			'available' => 0,
+		    );
+		    $types['tickets'] = array(
+			'count'     => 0, // count of tickets currently for sale
+			'stock'     => 0, // current stock of tickets available for sale
+			'global'    => 0, // global stock ticket
+			'unlimited' => 0, // unlimited stock tickets
+			'available' => 0, // are tickets available for sale right now
+		    );
+
+		    foreach ( $tickets as $ticket ) {
+			// If a ticket is not current for sale do not count it
+			if ( ! tribe_events_ticket_is_on_sale( $ticket ) ) {
+			    continue;
+			}
+
+			// if ticket and not rsvp add to ticket array
+			if ( 'Tribe__Tickets__RSVP' !== $ticket->provider_class ) {
+			    $types['tickets']['count'] ++;
+
+			    $global_stock_mode = $ticket->global_stock_mode();
+
+			    if ( Tribe__Tickets__Global_Stock::GLOBAL_STOCK_MODE === $global_stock_mode && 0 === $types['tickets']['global'] ) {
```

Comment:
This is a nice change and avoids unnecessary function calls - but we should 'yoda-ify' it, ie for the first part we should switch to:

`$global_stock_mode === Tribe__Tickets__Global_Stock::GLOBAL_STOCK_MODE`